source/agora/utils/TxBuilder.d

```diff
@@ -493,31 +559,44 @@ public struct TxBuilder
     /// Refund output for the transaction
     private Output leftover;
 
+    // fee per byte rate to be paid for the tx
+    private Amount fee_rate = Amount(700);
+
     /// Stores the inputs to consume until `sign` is called
     private OutputRef[] inputs;
 
     /// Transactions to be built and returned
     private Transaction data;
+
+    /// Unlocker delegate to sign the inputs
+    private Unlock delegate (in Transaction tx, in OutputRef out_ref)
+        @safe nothrow unlocker;
 }
 
 /// Test for a split with the same amount of outputs as inputs
-/// Essentially doing an equality transformation
+/// Essentially doing an equality transformation but with refund due to fee rate
 unittest
 {
     immutable Number = GenesisBlock.payments.front.outputs.length;
     assert(Number == 8);
 
-    const tx = TxBuilder(GenesisBlock.payments.front)
+    auto fee_rate = Amount(700);
+    const tx = TxBuilder(GenesisBlock.payments.front).feeRate(fee_rate)
         .split(WK.Keys.byRange.map!(k => k.address).take(Number))
         .sign();
 
-    // This transaction has 8 txs, hence it's just equality
+    // This transaction splits to 8 outputs but includes a refund
     assert(tx.inputs.length == Number);
-    assert(tx.outputs.length == Number);
+    assert(tx.outputs.length == Number + 1); // refund
     // Since the amount is evenly distributed in Genesis,
     // they all have the same value
-    const ExpectedAmount = genesisSpendable().front().leftover.value;
-    assert(tx.outputs.all!(val => val.value == ExpectedAmount));
+    auto fees = fee_rate * tx.sizeInBytes;
+    auto total = genesisSpendable().front().leftover.value * Number;
+    auto expectedAmount = (total - fees) / Number;
+    assert(tx.outputs.count!(o => o.value == expectedAmount) == Number);
+    auto sum = expectedAmount * Number + fees;
+    auto refund = total - sum;
+    assert(tx.outputs.count!(o => o.value == refund) == 1);
 }
 
```

Comment:
The rest of the tests in this module need to be updated similar to the one above.

Reply:
Will update the tests on a local branch.